### PR TITLE
Version 1.3 API updates

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** [#8]

**This PR will result in the following new package version:** Version 0.4.0

We are updating old fields and renaming them to current fields based on the API updates, and we ensured backward compatibility by coalescing the two fields.

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:** 
- In the [July 2023 connector update for TikTok Ads](https://fivetran.com/docs/applications/tiktok-ads/changelog), we upgraded the connector to the v1.3 API. As a result the dependent fields and field names from the downstream staging models have changed.
- In `stg_tiktok_ads__ad_group_history`, `age` has been renamed as `age_groups`.
- In `stg_tiktok_ads__advertiser`, `phone_number` has been renamed as `cellphone_number`, and `telephone` has been renamed as `telephone_number`.
- We coalesced the old API field names with these renamed fields to ensure backwards compatibility for these fields. 
  

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

Renamed fields were appropriately showing from local and seed runs. Running the latest version of package should reflect the renamed fields from the above models.

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [NA] DECISIONLOG updates have been updated (if applicable)
- [x] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🕐

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,16 @@
-# dbt_tiktok_ads_source v0.UPDATE.UPDATE
-
- ## Under the Hood:
-
+# dbt_tiktok_ads_source v0.4.0
+[PR #10](https://github.com/fivetran/dbt_tiktok_ads_source/pull/10) applies the following updates:
+## 🚨 Breaking Changes 🚨
+- In the [July 2023 connector update for TikTok Ads](https://fivetran.com/docs/applications/tiktok-ads/changelog), we upgraded the connector to the v1.3 API. As a result the dependent fields and field names from the downstream staging models have changed.
+- In `stg_tiktok_ads__ad_group_history`, `age` has been renamed as `age_groups`.
+- In `stg_tiktok_ads__advertiser`, `phone_number` has been renamed as `cellphone_number`, and `telephone` has been renamed as `telephone_number`.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: Small suggestion to make this read a bit clearer.
```suggestion
- In the [July 2023 connector update for TikTok Ads](https://fivetran.com/docs/applications/tiktok-ads/changelog), we upgraded the connector to the v1.3 API. As a result the dependent fields and field names from the downstream staging models have changed:
  - In `stg_tiktok_ads__ad_group_history`, `age` has been renamed as `age_groups`.
  - In `stg_tiktok_ads__advertiser`, `phone_number` has been renamed as `cellphone_number`, and `telephone` has been renamed as `telephone_number`.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+- We coalesced the old API field names with these renamed fields to ensure backwards compatibility for these fields.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:7]: This is more of a note as opposed to a breaking change. I would include it as a sub note to call out to users.
```suggestion
>**Note**: We coalesced the old API field names with these renamed fields to ensure backwards compatibility for these fields. 
```

File: models/stg_tiktok_ads__ad_group_history.sql [hunk 3]
```diff
@@ -29,7 +29,7 @@ final as (
         action_days,
         action_categories,
         adgroup_name as ad_group_name,
-        age,
+        coalesce(age, age_groups) as age_groups,
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/stg_tiktok_ads__ad_group_history.sql:32]: Can you apply this to the other coalesce statements in this PR.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-avinash thanks for working through this API update! Your changes look good, I just have a few requests to update the code highlighted in my review below.

Additionally, one more change that needs to be applied before approving:
- Be sure to update the README version of the package as this is a breaking change. I noticed you marked that check as `NA` in the PR submission form. However, since this is a breaking change, we will want to make that update to reference the new version range.

File: CHANGELOG.md (hunk 2 above)
Comment (fivetran-avinash) [reply CHANGELOG.md]: Updated

File: models/stg_tiktok_ads__ad_group_history.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/stg_tiktok_ads__ad_group_history.sql]: Done!

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz Thanks for noting the lack of a README update! Fixed that and applied all other changes now.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Changes look great! Thanks for working through these. Approved!

**Pull request merged**